aiida/orm/implementation/general/node.py

```diff
@@ -1604,6 +1653,64 @@ def __del__(self):
         if getattr(self, '_temp_folder', None) is not None:
             self._temp_folder.erase()
 
+
+    def get_hash(self, ignore_errors=True, **kwargs):
+        """
+        Making a hash based on my attributes
+        """
+        from aiida.common.hashing import make_hash
+        try:
+            return make_hash(self._get_objects_to_hash(), **kwargs)
+        except Exception as e:
+            if ignore_errors:
+                return None
+            else:
+                raise e
+
+    def _get_objects_to_hash(self):
+        """
+        Return a list of objects which should be included in the hash.
+        """
+        computer = self.get_computer()
+        return [
+            importlib.import_module(
+                self.__module__.split('.', 1)[0]
+            ).__version__,
+            {
+                key: val for key, val in self.get_attrs().items()
+                if (
+                    (key not in self._hash_ignored_attributes) and
+                    (key not in getattr(self, '_updatable_attributes', tuple()))
+                )
+            },
+            self.folder,
+            computer.uuid if computer is not None else None
+        ]
+
+    def rehash(self):
+        self.set_extra('hash', self.get_hash())
+
+    def get_same_node(self):
+        if not self._cacheable:
```

Comment:
Yes, that's exactly what it does. I added a docstring, and since this function is only used internally, I changed the name with a leading underscore.

Also, I moved the implementation to ``_iter_all_same_nodes`` which returns an iterator over all same nodes. The advantage is that I can re-use the same code for the ``get_all_same_nodes`` you propose below. I didn't use ``get_all_same_nodes`` directly because then I would have to evaluate the full query (``.all()`` instead of ``.iterall()``) also for the case when only one match is needed.